samples/nrf9160/http_update/src/update.c

```diff
@@ -0,0 +1,256 @@
+#include <drivers/gpio.h>
+#include <drivers/flash.h>
+#include <bsd.h>
+#include <modem/lte_lc.h>
+#include <modem/at_cmd.h>
+#include <modem/at_notif.h>
+#include <modem/bsdlib.h>
+#include <modem/modem_key_mgmt.h>
+#include <net/fota_download.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+#include <stdio.h>
+
+#define LED_PORT DT_GPIO_LABEL(DT_ALIAS(led0), gpios)
+#define TLS_SEC_TAG 42
+
+static const struct device *gpiob;
+static struct gpio_callback gpio_cb;
+static struct k_work fota_work;
+
+void (*finished_cb)(void);
+const char *(*host_get)(void);
+const char *(*file_get)(void);
+bool (*two_leds)(void);
+
+#ifndef CONFIG_USE_HTTPS
+#define SEC_TAG (-1)
+#else
+#define SEC_TAG (TLS_SEC_TAG)
+#endif
+
+/**@brief Recoverable BSD library error. */
+void bsd_recoverable_error_handler(uint32_t err)
+{
+	printk("bsdlib recoverable error: %u\n", err);
+}
+
+int cert_provision(void)
+{
+	static const char cert[] = {
+#include "../cert/BaltimoreCyberTrustRoot"
+	};
+	BUILD_ASSERT(sizeof(cert) < KB(4), "Certificate too large");
+	int err;
+	bool exists;
+	uint8_t unused;
+
+	err = modem_key_mgmt_exists(TLS_SEC_TAG,
+				    MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, &exists,
+				    &unused);
+	if (err) {
+		printk("Failed to check for certificates err %d\n", err);
+		return err;
+	}
+
+	if (exists) {
+		/* For the sake of simplicity we delete what is provisioned
+		 * with our security tag and reprovision our certificate.
+		 */
+		err = modem_key_mgmt_delete(TLS_SEC_TAG,
+					    MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN);
+		if (err) {
+			printk("Failed to delete existing certificate, err %d\n",
+			       err);
+		}
+	}
+
+	printk("Provisioning certificate\n");
+
+	/*  Provision certificate to the modem */
+	err = modem_key_mgmt_write(TLS_SEC_TAG,
+				   MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert,
+				   sizeof(cert) - 1);
+	if (err) {
+		printk("Failed to provision certificate, err %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+/**@brief Start transfer of the file. */
+static void update_transfer_start(struct k_work *unused)
+{
+	int retval;
+	char *apn = NULL;
+
+	/* Functions for getting the host and file */
+	retval = fota_download_start(host_get(), file_get(), SEC_TAG, apn, 0);
+	if (retval != 0) {
+		/* Re-enable button callback */
+		gpio_pin_interrupt_configure(gpiob,
+					     DT_GPIO_PIN(DT_ALIAS(sw0), gpios),
+					     GPIO_INT_EDGE_TO_ACTIVE);
+
+		printk("fota_download_start() failed, err %d\n", retval);
+		return;
+	}
+}
+
+static int led_init(void)
+{
+	const struct device *dev;
+
+	dev = device_get_binding(LED_PORT);
+	if (dev == 0) {
+		printk("Nordic nRF GPIO driver was not found!\n");
+		return 1;
+	}
+
+	gpio_pin_configure(dev, DT_GPIO_PIN(DT_ALIAS(led0), gpios),
+			   GPIO_OUTPUT_ACTIVE |
+				   DT_GPIO_FLAGS(DT_ALIAS(led0), gpios));
+
+	if (two_leds()) {
+		gpio_pin_configure(dev, DT_GPIO_PIN(DT_ALIAS(led1), gpios),
+				   GPIO_OUTPUT_ACTIVE |
+					   DT_GPIO_FLAGS(DT_ALIAS(led1),
+							 gpios));
```

Comment:
its clang-format, I'll modify it manually then.

Reply:
Same for lines [119](https://github.com/nrfconnect/sdk-nrf/blob/7e0bcf318493704d3f107ca14da30d580ee0bde7/samples/nrf9160/http_update/src/update.c#L119) and [150](https://github.com/nrfconnect/sdk-nrf/blob/7e0bcf318493704d3f107ca14da30d580ee0bde7/samples/nrf9160/http_update/src/update.c#L150).

Reply:
So should we go in and manually modify these, or should we say that whatever clang-format outputs is acceptable?

to me the indendation clang-format outputs (ie the current one) is the most readabel (without extracting to variables, which is a bit hard since these are nested #defines)

Reply:
To answer mayself, what we should do in situations like this is to modify the code so that clang-format will actually generate good-looking code. Which I have done now :)